unistd/pwd.c

```diff
@@ -24,69 +24,94 @@
 
 struct passwd pwnam;
 char pw_name[NAME_MAX];
+char intstr[16];
 char pw_gecos[128];
 char pw_dir[PATH_MAX];
 char pw_shell[PATH_MAX];
 char pw_passwd[128];
 
 
-struct passwd *getpwnam(const char *name)
+static void readpwentry(FILE *fp, char *pwentry, unsigned int maxlgth)
 {
-	FILE *fp;
-	char buf[1024];
+	char c = 0;
+	unsigned int i = 0;
+	c = fgetc(fp);
+	while (c != ':' && c != '\n' && c != EOF) {
+		if (i < maxlgth - 1)
+			pwentry[i++] = c;
+		c = fgetc(fp);
+	}
```

Comment:
Oh, yeah, when we exit because `i` got too big, we'll need to purge to next delimiter (or even better, exit with error).

Reply:
In your suggestion you probably meant:
```
(c == ':' || c == '\n' || c == EOF) 
```
instead of:
```
(c != ':' || c != '\n' || c == EOF) 
```
:) Isn`t one additional `fgetc` call better than whole purging `while`?

Reply:
Yeah, you're right, I haven't inverted other part of the logic. I don't understand your question, though.

Reply:
not doing a careful review, just pointing out some thins noticed in email thread:

probably `maxlen` variable name would be a better fit :)

Reply:
Implemented `maxlen` suggestion. Thanks.

Overall i implemented your suggestion :) If we purge the whole file based on one bad entry then we do not need to purge to next delimiter. However i added `(i < maxlgth - 1)` check into the while and terminate it based on `isbad` variable.